src/main/java/com/jcabi/aspects/aj/ImprovedJoinPoint.java

```diff
@@ -32,47 +32,103 @@
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
 import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.Signature;
 import org.aspectj.lang.reflect.MethodSignature;
+import org.aspectj.lang.reflect.SourceLocation;
 
 /**
- * Utility methods that deal with JoinPoints.
+ * This class implements the methods from JointPoint interface.
  *
- * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
+ * @author Shelan Perera (shelanrc@gmail.com)
  * @version $Id$
- * @since 0.7.22
+ * @since 1.0
  */
-final class JoinPointUtils {
+public class ImprovedJoinPoint implements JoinPoint {
 
     /**
-     * Utility class constructor.
+     *
      */
-    private JoinPointUtils() {
-        // intentionally left empty
+    private final transient JoinPoint joinpoint;
+
+    /**
+     * Constructor for BaseJointPoint.
```

Comment:
@shelan why `BaseJointPoint`?
anyway  I think `Constructor.` will be enough or even `Ctor.`